File: 2021/10/04/main.go
package main

var dx = []int{-1, 1, 0, 0}
var dy = []int{0, 0, -1, 1}

func islandPerimeter(grid [][]int) int {
	r, m, n := 0, len(grid), len(grid[0])
	for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if grid[i][j] == 0 {
				continue
			}
			r += 4
			for k := 0; k < 4; k++ {
				x := i + dx[k]
				y := j + dy[k]
				if 0 <= x && x < m && 0 <= y && y < n && grid[x][y] == 1 {
					r--
				}
			}
		}
	}

	return r
}

File: 2021/10/04/main.py
class Solution:
    def islandPerimeter(self, grid: List[List[int]]) -> int:
        m, n = len(grid), len(grid[0])
        dx = [-1, 1, 0, 0]
        dy = [0, 0, -1, 1]

        r = 0
        for i in range(m):
            for j in range(n):
                if grid[i][j] == 0:
                    continue
                r += 4
                for k in range(4):
                    x = i + dx[k]
                    y = j + dy[k]
                    if 0 <= x and x < m and 0 <= y and y < n and grid[x][y] == 1:
                        r -= 1
        return r